_episodes_rmd/06_day5.Rmd

```diff
@@ -0,0 +1,217 @@
+
+```{r load package}
+
+library(raster)
+library(sf)
+library(ggplot2)
+
+```
+
+
+## Raster data
+For this example, we will use the World Climate dataset. You can either download this dataset from the website. Or you can access the dataset using the R package.
+
+If you need a refresher on how to do this, look at this website: https://www.gis-blog.com/r-raster-data-acquisition/ 
+
+The goal of this section is to have your raster data as a 'Formal class RasterLayer' in your Global Environment
+
+```{r load-raster-data}
+
+# get worldclim data, variable "bio" with a 10 degree resolution
+# this line is downloading data
+# if you selected a resolution higher than 10 (with a res of 0.5,2.5, or 5) it may take a while to download
+worldClim_bio_global <- getData("worldclim", var="bio", res=10)
+
+# use only the bio5 variable (BIO5 = Max Temperature of Warmest Month)
+plot(worldClim_bio_global$bio5)
+
+maxTemp_global <- worldClim_bio_global$bio5
+
+
+```
+
+
+## Vector data
+For this example, we will use the Natual Earth Populated Places dataset. You can access these data from the Natural Earth package, or download the data from the website.
+
+The goal of this section is to have a simple feature ('sf') data frame type of object in your Global Environment.
+
+
+```{r load-city-point-data}
+
+# import point data from a csv file
+WIcities <- read.csv("data/ne_10m_populated_places_Wisconsin.csv")
+
+# check out the data
+colnames(WIcities)
+
+```
+
+Recall from your lesson how to import point data and convert it to a spatial data layer in R.
+
+When looking at the column names we see that there is a lot of data! To convert to a spatial points data layer, we need to know information about the coordinate system and the coordinates for each point.
+
+The column names 'LATITUDE' and 'LONGITUDE' have what we need! 
+
+Explain how we know about the coordinate system because of these coordinates.
+Reference this website: https://spatialreference.org/ref/epsg/wgs-84/ 
+
+> ## Solution: Load .csv point data?
+>
+> Text here describing what's in this box
+>
+>
+> > ## Solution
+> >
+> > ```{r, echo=FALSE}
+> > paste("This", "new", "template", "looks", "good")
+> > ```
+> {: .solution}
+
+```{r convert-to-points}
+
+# convert data framem to sf type of object by specifying coordinates (as xy) and the projection as a crs
+WIcities_points <- st_as_sf(WIcities, coords=c('LONGITUDE','LATITUDE'), crs=crs(maxTemp_global))
+
+# check class of object - should be "sf" data frame
+class(WIcities_points)
+
+
+```
+
+A critical part of working with raster and vector data together is that the dataasets are in the same coordinate system. 
+
+Does it make sense? Are they the same?
+
+```{r check-coordinate-systems}
+
+crs(maxTemp_global)
+crs(WIcities_points)
+
+crs(maxTemp_global)
+crs(WIcities_points)
+
+extent(maxTemp_global)
+extent(WIcities_points)
+
+res(maxTemp_global)
+
+```
```

Comment:
I really like this part of the lesson looking at crs and extent - it re-emphasizes a lot of the stuff covered in the lessons!